pkg/cloud/services/s3/s3.go

```diff
@@ -97,6 +97,47 @@ func (s *Service) DeleteBucket() error {
 
 	log.Info("Deleting S3 Bucket")
 
+	// Delete machine pool user data files that did not get deleted
+	// yet by the lifecycle policy
+	for {
```

Comment:
The `machinepool` feature is a feature that needs to be enabled using feature flags. We should probably check the feature flag before executing this logic.